c7n/resources/ec2.py

```diff
@@ -206,6 +206,34 @@ def get_resource_date(self, i):
         return None
 
 
+class ASGFilter(object):
+    """Filter instances by whether they are members of an ASG"""
+    asg_member = None
+
+    def filter_asg_membership(self, instances, desired=None):
+        if desired is None:
+            desired = self.asg_member
+
+        if self.asg_member is None:
+            # TODO raise exception
```

Comment:
its fine if the autorecover alarm inlines the filter (either composition or direct), i'm just cautious about including a separate filter for this, vs. just documenting the syntax for using the value type to filter correctly in the docs for the autorecover alarm action.